Plugins/NativeEngine/Source/NativeEngine.h

```diff
@@ -43,11 +43,26 @@ namespace Babylon
     struct ProgramData final
     {
         ProgramData() = default;
-        ProgramData(ProgramData&& other) = delete;
         ProgramData(const ProgramData&) = delete;
-        ProgramData& operator=(ProgramData&& other) = delete;
         ProgramData& operator=(const ProgramData& other) = delete;
 
+        ProgramData(ProgramData&& other) noexcept
+        {
+            (*this) = std::move(other);
+        }
+
+        ProgramData& operator=(ProgramData&& other) noexcept
+        {
+            Handle = std::move(other.Handle);
+            other.Handle = bgfx::ProgramHandle{bgfx::kInvalidHandle};
+            Disposed = std::move(other.Disposed);
+            Uniforms = std::move(other.Uniforms);
+            UniformNameToIndex = std::move(other.UniformNameToIndex);
+            UniformInfos = std::move(other.UniformInfos);
+            VertexAttributeLocations = std::move(other.VertexAttributeLocations);
+            return *this;
```

Comment:
Assuming we remove `Disposed`:
```suggestion
            Handle = other.Handle;
            other.Handle = BGFX_INVALID_HANDLE;
            Uniforms = std::move(other.Uniforms);
            UniformNameToIndex = std::move(other.UniformNameToIndex);
            UniformInfos = std::move(other.UniformInfos);
            VertexAttributeLocations = std::move(other.VertexAttributeLocations);
            return *this;
```